pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -534,10 +707,19 @@ func afwGetReleventStatefulsetByKind(cr splcommon.MetaObject, client splcommon.C
 // afwSchedulerEntry Starts the scheduler Pipeline with the required phases
 func afwSchedulerEntry(client splcommon.ControllerClient, cr splcommon.MetaObject, appDeployContext *enterpriseApi.AppDeploymentContext, appFrameworkConfig *enterpriseApi.AppFrameworkSpec) error {
 	scopedLog := log.WithName("afwSchedulerEntry").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
-	afwPipeline = initAppInstallPipeline()
+	afwPipeline = initAppInstallPipeline(appDeployContext)
 
 	var clusterScopedApps []*enterpriseApi.AppDeploymentInfo
 
+	afwEntryTime := time.Now().Unix()
+
+	// get the current available disk space for downloading apps on operator pod
+	availableDiskSpace, err := getAvailableDiskSpace()
+	if err != nil {
```

Comment:
Now that we have a return here, can you please move the defer function at line 817 here, so that we are sure that all the memory is released? I should have done it in my original PR, but seems like I missed it.

Reply:
Yes that defer function is too far below, needs to be somewhere in the beginning of this function.